libnopegl/src/internal.h

```diff
@@ -702,6 +702,9 @@ void *ngli_node_get_data_ptr(struct ngl_node *var_node, void *data_fallback);
 int ngli_prepare_draw(struct ngl_ctx *s, double t);
 void ngli_node_draw(struct ngl_node *node);
 
+typedef int (*ngli_children_func_type)(void *user_arg, struct ngl_node *node, struct ngl_node *child);
+int ngli_node_children_apply_func(ngli_children_func_type func, void *user_arg, struct ngl_node *node);
```

Comment:
Can we have the argument in the following order (and node renamed to parent for the func type):
typedef int (*ngli_children_func_type)(struct ngl_node *parent, struct ngl_node *child, void *user_arg);`
`int ngli_node_children_apply_func(struct ngl_node *node, ngli_children_func_type func, void *user_arg);`
?

Reply:
Also maybe renamed the function to `ngli_node_children_func_type` ?

Reply:
- Fixed the node/child vs parent/node confusion
- `user_arg` is usually first in the codebase (rationale being that it's a context)
- I don't understand the `ngli_node_children_func_type` function rename; `_func_type` refers to a typedef

Reply:
Thanks. Regarding `user_arg` I got confused by the fact that we placed `user_arg` in the recently introduced callbacks setters (see `ngli_darray_set_free_func`, `ngli_darray_set_free_func`).

Regarding the rename it's about: `ngli_children_func_type` -> `ngli_node_children_func_type`.

Reply:
The function is renamed to `children_apply_func` so I guess I can ignore that? :D

Reply:
Ok.